src/api/helpers.py

```diff
@@ -0,0 +1,329 @@
+                                "info": existing_dataset.info,
+                                "omop_table": "dataset",
+                            }
+                        )
+                    else:
+                        # Create new dataset
+                        new_dataset = await create_record(
+                            session, id_mapper, field, "dataset"
+                        )
+                        new_dataset_id = new_dataset["id"]
+                        new_dataset["omop_table"] = "dataset"
+                        return_obj.append(new_dataset)
+                    break
+
+            if new_dataset_id is None:
+                await raise_bad_request("dataset")
+
+            new_person_id = None
+            for field in items:
+                if field.get("omop_table") == "person" and not field.get("skip_errors"):
+                    new_person = await create_record(
+                        session, id_mapper, field, "person"
+                    )
+                    new_person_id = new_person["person_id"]
+                    return_obj.append(new_person)
+                    break
+
+            if new_person_id is None:
+                await raise_bad_request("person")
+
+            # Second pass: Process all other tables at the donor level
+            for field in items:
+                table_name = field.get("omop_table")
+                if (
+                    table_name != "person"
+                    and table_name != "dataset"
+                    and table_name in TABLE_CONFIG
+                    and not field.get("skip_errors")
+                ):
+                    new_record = await create_record(
+                        session, id_mapper, field, table_name
+                    )
+                    return_obj.append(new_record)
+
+        else:  # process other tables
+            # Check if the key matches any patterns (excluding donor)
+            is_known_pattern = any(
+                re.match(pattern, key)
+                for name, pattern in PATTERNS.items()
+                if name not in ["donor", "dataset"]
+            )
+            if is_known_pattern:
+                for field in items:
+                    table_name = field.get("omop_table")
+                    if table_name in TABLE_CONFIG and not field.get("skip_errors"):
+                        new_record = await create_record(
+                            session, id_mapper, field, table_name
+                        )
+                        return_obj.append(new_record)
+
+
+async def handle_single_donor_data_ingestion(body: dict):
+    """
+    Ingest single donor's data structure.
+    """
+    return_obj = []
+
+    async for session in get_db_session():
+        try:
+            await ingest_donor_with_clinical_data(session, body, return_obj)
+            await session.commit()
+            logger.info("Successfully created all records and committed transaction.")
```

Comment:
```suggestion
            logger.info("Successfully created records and committed transaction for a donor.")
```
And further edit it with which donor, if possible to get that here

Reply:
ok. For which donor gets created, I'll add a TODO there, since I don't want to return the whole donor record, only the id